src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1003,6 +1003,14 @@ parameters[i].RefKind switch
 
         private void EmitArrayElementLoad(BoundArrayAccess arrayAccess, bool used)
         {
+            if (arrayAccess.IsRef)
```

Comment:
>if (arrayAccess.IsRef) [](http://example.com/codeflow?start=12&length=22)

I think it would be better to add a special node wrapping ```BoundArrayAccess``` rather than adding a new property with unclear effect that could be set improperly. The code in the `if` would go into emit helper for the new node.